velox/functions/prestosql/tests/GreatestLeastTest.cpp

```diff
@@ -181,3 +190,97 @@ TEST_F(GreatestLeastTest, clearNulls) {
     }
   }
 }
+
+TEST_F(GreatestLeastTest, shortDecimal) {
+  const auto shortType = DECIMAL(10, 4);
+  const std::vector<std::optional<int64_t>> firstColumnUnscaled = {
+      10000,
+      -10000,
+      20000,
+      UnscaledShortDecimal::max().unscaledValue() - 1,
+      UnscaledShortDecimal::min().unscaledValue() + 1,
+      std::nullopt};
+  const std::vector<std::optional<int64_t>> secondColumnUnscaled = {
+      -10000,
+      10000,
+      -20000,
+      UnscaledShortDecimal::min().unscaledValue() + 1,
+      UnscaledShortDecimal::max().unscaledValue() - 1,
+      1};
+  const auto firstColumn =
+      makeNullableShortDecimalFlatVector(firstColumnUnscaled, shortType);
+  const auto secondColumn =
+      makeNullableShortDecimalFlatVector(secondColumnUnscaled, shortType);
+
+  const auto expectedLeast = makeNullableShortDecimalFlatVector(
+      {-10000,
+       -10000,
+       -20000,
+       UnscaledShortDecimal::min().unscaledValue() + 1,
+       UnscaledShortDecimal::min().unscaledValue() + 1,
+       std::nullopt},
+      shortType);
+  const auto expectedGreatest = makeNullableShortDecimalFlatVector(
+      {10000,
+       10000,
+       20000,
+       UnscaledShortDecimal::max().unscaledValue() - 1,
+       UnscaledShortDecimal::max().unscaledValue() - 1,
+       std::nullopt},
+      shortType);
+
+  runDecimalTest<UnscaledShortDecimal>("least(c0)", {firstColumn}, firstColumn);
```

Comment:
Template parameter is not needed. Let's remove it. Consider renaming firstColumn and secondColumns to a and b for readability. Also, consider organizing the test like so:

```
runDecimalTest("least(c0)", {a}, a);

auto expected = ...
runDecimalTest("least(c0, c1)", {a, b}, expected);

runDecimalTest("greatest(c0)", {a}, a);

expected = ...
runDecimalTest("greatest(c0, c1)", {a, b}, expected);
```

Reply:
The tests' organization is changed as suggested. Could you please share why the template variable is not needed? It helps improve readability by having a common ```runDecimalTest``` function.